File: lib/scripts/replace_page_604.dart
import 'dart:io';

void main() async {
  // Read the original file
  final originalFile = File('D:\\solutions\\MuslimLifeAI_demo\\assets\\quran\\mushaf_v2_map.txt');
  final lines = await originalFile.readAsLines();
  
  // Read the repaired lines
  final repairedFile = File('D:\\solutions\\MuslimLifeAI_demo\\page_604_repaired.txt');
  final repairedLines = await repairedFile.readAsLines();
  
  // Replace Page 604 lines
  final newLines = <String>[];
  bool skip604 = false;
  
  for (final line in lines) {
    if (line.startsWith('604,')) {
      if (!skip604) {
        // Add all repaired lines
        newLines.addAll(repairedLines);
        skip604 = true;
      }
      // Skip original 604 lines
    } else if (line.startsWith('605,')) {
      skip604 = false;
      newLines.add(line);
    } else {
      newLines.add(line);
    }
  }
  
  // Write back
  await originalFile.writeAsString(newLines.join('\n'));
  
  print('✅ Successfully replaced Page 604 lines in mushaf_v2_map.txt');
  print('   Replaced 15 lines with ${repairedLines.length} repaired lines');
}

File: lib/scripts/replace_page_604.py
#!/usr/bin/env python3
"""Replace Page 604 lines in mushaf_v2_map.txt with repaired lines"""

def main():
    # Read the original file
    with open(r'D:\solutions\MuslimLifeAI_demo\assets\quran\mushaf_v2_map.txt', 'r', encoding='utf-8') as f:
        lines = f.readlines()
    
    # Read the repaired lines
    with open(r'D:\solutions\MuslimLifeAI_demo\page_604_repaired.txt', 'r', encoding='utf-8') as f:
        repaired_lines = [line.strip() + '\n' for line in f.readlines()]
    
    # Replace Page 604 lines
    new_lines = []
    skip_604 = False
    
    for line in lines:
        if line.startswith('604,'):
            if not skip_604:
                # Add all repaired lines
                new_lines.extend(repaired_lines)
                skip_604 = True
            # Skip original 604 lines
        elif line.startswith('605,'):
            skip_604 = False
            new_lines.append(line)
        else:
            new_lines.append(line)
    
    # Write back
    with open(r'D:\solutions\MuslimLifeAI_demo\assets\quran\mushaf_v2_map.txt', 'w', encoding='utf-8') as f:
        f.writelines(new_lines)
    
    print('✅ Successfully replaced Page 604 lines in mushaf_v2_map.txt')
    print(f'   Replaced 15 lines with {len(repaired_lines)} repaired lines')

if __name__ == '__main__':
    main()
